src/pages/registration/resource_type_tab/components/RevisionField.tsx

```diff
@@ -0,0 +1,35 @@
+import { useTranslation } from 'react-i18next';
+import { Field, FieldProps, useFormikContext } from 'formik';
+import { ResourceFieldNames } from '../../../../types/publicationFieldNames';
+import { Checkbox, FormControlLabel, InputProps } from '@mui/material';
+import { BookRegistration, Revision } from '../../../../types/publication_types/bookRegistration.types';
+
+export const RevisionField = () => {
+  const { setFieldValue } = useFormikContext<BookRegistration>();
+  const { t } = useTranslation();
+
+  const handleRevisionChange = (checked: boolean, field: InputProps) => {
+    setFieldValue(ResourceFieldNames.Revision, checked ? Revision.REVISED : Revision.UNREVISED);
+  };
+
+  return (
+    <Field name={ResourceFieldNames.Revision}>
+      {({ field }: FieldProps) => (
+        <FormControlLabel
+          label={t('registration.is_revision')}
+          control={
+            <Checkbox
+              checked={Revision.REVISED === field.value}
+              inputProps={{ 'aria-label': t('registration.is_revision') }}
+              id="book-revision-checkbox"
+              {...field}
+              onChange={(_event, checked) => {
+                handleRevisionChange(checked, field);
```

Comment:
om man skal pirke ekstra (siden du ville ha slakt(?) 😅), så kan `setFieldValue` godt legges direkte hit. Forskjellen da vil være at `handleRevisionChange` ikke vil reinitialiseres for hver render